spec/features/course_management_spec.rb

```diff
@@ -8,30 +8,28 @@
     let!(:user) { create(:user, role: :administrator) }
     before { login_as(user, scope: :user) }
 
-    describe 'course new page' do
-      before { visit new_course_path }
+    scenario 'Users can see a list of courses' do
+      create(:course, creator: user, updater: user)
 
-      it { is_expected.to have_field('course_title') }
-      it { is_expected.to have_field('course_description') }
+      visit courses_path
+      expect(all('.course').count).to eq(1)
+      expect(subject).to have_link(I18n.t('course.courses.index.new'), href: new_course_path)
     end
 
-    describe 'course creation' do
-      before { visit new_course_path }
-      subject { click_button I18n.t('helpers.submit.course.create') }
+    scenario 'Users can create a new course' do
```

Comment:
OK, But there will always be some branches in feature tests, how to deal with that ?  

And look at capybara's feature test examples, they also use `scenario` to deal with such things:
https://github.com/jnicklas/capybara#using-capybara-with-rspec

My main concern is that currently our testing styles are inconsistent, maybe we should discuss this and put it in the guideline.